diesel/src/pg/metadata_lookup.rs

```diff
@@ -49,9 +49,18 @@ impl PgMetadataLookup {
     }
 }
 
-fn lookup_type(
+/// Gets the [`PgMetadataCache`] for a `Connection<Backend=Pg>`
+/// so that the lookup of user defined types, or types which come from an extension can be cached.
+///
+/// Implementing this trait for a `Connection<Backend=Pg>` will cause [`PgMetadataLookup`] to be auto implemented.
+pub trait GetPgMetadataCache {
```

Comment:
That trait needs to be reexported via in `crate::pg` similarly to the `PgMetadataLookup` trait itself.

Reply:
What version of rust are the docs that actually get published built with. Locally I don't get an error because version 1.40 doesn't doesn't understand the syntax. I just want to make sure the published docs will link correctly.

Reply:
Also if I reexport `GetPgMetadataCache`, docs fail because of the doc comment on it refers to `PgMetadataCache` so I would have to reexport that too. If we reexport `PgMetadataCache`, I would like to make `PgMetadataCache::new` public. I don't know if y'all want to make all of this public or not. If y'all don't, is there a way to hide part of a doc comment? Because that comment is still useful for contributors.

Reply:
Docs are build with the latest stable release, but I would expect that building docs works for all supported rust versions.
As additional note: If someone (for example) want's to use this in a third party crate this (`GetPgMetadataCache` + PgMetadataCache`) needs to be "public". As already noted in our gitter channel, I would like to have a general policy how to deal with public API for third-party backend/connections. Probably something like: We do guarantee there is no breaking change for this "special" part of the API for patch releases. We try to minimize the breakage between minor releases. + Opt in via a feature flag + Some explicit documentation about this (and not including this documentation in published docs). But that needs some discussion about which API should be included and if that's even the right approach. 
For the mean time I'm fine with having those items as `#[doc(hidden)]` which excludes them from our public API (so no guarantees for any user).

Reply:
Looks like using type links is error prone without this special syntax ([rust blog](https://blog.rust-lang.org/2020/11/19/Rust-1.48.html)) especially since we reexport a lot. Should I just drop the link?

Reply:
Droping this link is fine for fixing this error, as most users don't actually need to interact with this trait at all. If someone really wants to implement a custom connection, it should be fine to do a bit more research on this topic and cpoy&past the type name into the search.